docker/Makefile

```diff
@@ -77,9 +77,9 @@ test: prep
 		rspec --options puppetserver/.rspec spec
 
 push-image: prep
-	@docker push puppet/puppetserver:$(VERSION)
```

Comment:
similar to my r10k comment, what is the rationale for doing this in the `push-image` target vs the `build` target?

Reply:
[@justinstoller](https://github.com/justinstoller)
Thanks for your response,
Here, we are using **buildx build** for building multiarchitecture images, where we need to use `--push` flag with buildx in makefile. 
If we want to build for single platform at a time, we can use `--load` flag to build and then push later.

Reply:
The Makefile already calls `docker buildx build` - new targets should be added there ... this make target is specifically for pushing.

Reply:
@Iristyle
```The Makefile already calls docker buildx build - new targets should be added there ... this make target is specifically for pushing.```

We need to use `–push` flag here instead of `--load`, as it needs to push multi-architecture images at a time, and we can use `--load` when we need to build an image for a specific platform, then we can push image separately for that specific architecture.

Reply:
My point is that we still want completely separate `build` and `push` targets in the Makefile. Since the `build` target wasn't modified to produce multiple architecture outputs, then this `push-image` target is also building, which is against the design of the Makefile